File: 1537-maximum-score-after-splitting-a-string/1537-maximum-score-after-splitting-a-string.py
class Solution:
    def maxScore(self, s: str) -> int:
        
        result=0
        for i in range(1,len(s)):
            count=0
            for j in s[:i]:
                if '0'==j:
                    count+=1
            for j  in s[i:]:
                if "1"==j:
                    count+=1
            result=max(count,result)
        return result

        

File: 1537-maximum-score-after-splitting-a-string/1537-maximum-score-after-splitting-a-string.java
class Solution {
    public int maxScore(String s) {
        int result=0;
        int count = 0;
        int left =0;
        int right = 0;
        for(int i = 0;i<s.length();i++)
        {
            if(s.charAt(i)=='0')
            {
                left++;
            }
        }
        for(int i = s.length()-1;i>0;i--)
        {
            if(s.charAt(i)=='1')
            {
                right++;
            }
            else
            {
                left--;
            }
            count=left+right;
            if(result<count)
            {
                result=count;
            }
        }
        return result;
        
    }
}